src/parseGithubMetadata.ts

```diff
@@ -1,69 +1,89 @@
 import matter from 'gray-matter'
+import validUrl from 'valid-url'
 
-import { PollMetadata, PollWithRawMetadata } from './polls'
-import { PollVoteType } from './polls'
-import { POLL_VOTE_TYPE, PollInputFormat } from './constants'
+import {
+  PollMetadata,
+  PollOptions,
+  PollWithRawMetadata,
+  PollVoteType,
+} from './polls'
+import { POLL_VOTE_TYPE } from './constants'
 import { assignTags } from './fetchPollTags'
+import {
+  validatePollParameters,
+  oldVoteTypeToNewParameters,
+} from './validatePollParameters'
+import { markdownToHtml } from './utils'
 
-export function parseGithubMetadata(
+export async function parseGithubMetadata(
   pollsWithRawMetadata: PollWithRawMetadata[],
   pollTagsFilePath: string
-): PollMetadata[] {
-  const polls = pollsWithRawMetadata
-    .map(({ pollId, rawMetadata }) => {
-      const { data: pollMetadata } = matter(rawMetadata)
+): Promise<PollMetadata[]> {
+  const polls = await Promise.all(
+    pollsWithRawMetadata.map(async ({ rawMetadata, ...poll }) => {
+      const { data: pollMetadata, content: markdownContent } =
+        matter(rawMetadata)
 
-      const title: string = pollMetadata.title || ''
+      const content = await markdownToHtml(markdownContent)
+
+      const title: string = pollMetadata?.title || ''
+      const summary: string = pollMetadata?.summary || ''
+      const options: PollOptions = pollMetadata?.options || {}
+      const discussionLink =
+        pollMetadata?.discussion_link &&
+        validUrl.isUri(pollMetadata.discussion_link)
+          ? pollMetadata.discussion_link
+          : null
+
+      // Parse old vote type
       const voteType: PollVoteType =
         (pollMetadata as { vote_type: PollVoteType | null })?.vote_type ||
         POLL_VOTE_TYPE.UNKNOWN
 
-      const pollType = pollMetadata.parameters
-        ? validatePollType(pollMetadata.parameters)
-        : oldVoteTypeToNew(voteType)
+      const [parameters, errorParameters] = pollMetadata.parameters
+        ? validatePollParameters(pollMetadata.parameters)
+        : oldVoteTypeToNewParameters(voteType)
+
+      if (errorParameters.length > 0 || !parameters) {
+        throw new Error(
+          `Invalid poll parameters for poll ${poll.pollId}. ${errorParameters}`
+        )
+      }
+
+      let startDate, endDate
+      //poll coming from poll create page
```

Comment:
why is this necessary? Are you saying polls coming from poll create page don't have the start and end date set posted on-chain?

Reply:
Actually, I recycled that part from the parsing metadata function from the Gov Portal, which can be called directly from the create page, passing a partial poll with 0 startDate and endDate. But in this GitHub action, that code should never be reachable as polls coming from Spock should always have start and end dates.

Reply:
Got it, we could remove the code or update the comment then just for readability

Reply:
Done, removed the code to always use start and end dates from Spock, gave a quick check to the aggregated file and it looks good, this is [the file](https://raw.githubusercontent.com/hernandoagf/community/master/governance/polls/meta/polls.json) in case you want to check it out